### PR TITLE
rel notes 05-04

### PR DESCRIPTION
**Pull request by kaarthis**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+* Starting with 1.24 the default format of clusterUser credential for AAD enabled clusters will be ‘exec’, which requires [kubelogin](https://github.com/Azure/kubelogin) binary in the execution PATH. If you are using Azure CLI, it will prompt users to download kubelogin. There will be no behavior change for non-AAD clusters, or AAD clusters whose version is older than 1.24. Existing downloaded kubeconfig will still work. We provide an optional query parameter ‘format’ when getting clusterUser credential to overwrite the default behavior change, you can explicitly specify format to ‘azure’ to get old format kubeconfig.
+* Konnectivity rollout will continue in May 2022 and is expected to complete by end of May.
+* Update your AKS labels to the recommended substitutions before deprecation after the Kubernetes v1.24 release. See more information on label deprecations and how to update your labels in the [Use labels in an AKS cluster](https://docs.microsoft.com/azure/aks/use-labels) documentation.
+
+### Release notes
+* Public preview
```
Comment (palma21) [inline CHANGELOG.md:14]: This is hardly a feature, more of a behavior change of the CLI with no arguments

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+  * The `aks-preview` Azure CLI extension (version 0.5.66+) now supports running `az aks update -g <resourceGroup> -n <clusterName>` without any optional arguments. This will perform an update operation without performing any changes, which can recover a cluster stuck in a failed provisioning state.
+  * [AKS now supports updating kubelet on node pools to use a new or changed user-assigned managed identity.](https://docs.microsoft.com/azure/aks/use-managed-identity#update-an-existing-cluster-using-kubelet-identity-preview)
+* Behavioral changes
+    * Fixes a bug with the AKS-EnableDualStack preview feature that would delete managed outbound IPv6 IPs if updating the cluster with a version of the API before the dual-stack parameters were added.
```
Comment (palma21) [inline CHANGELOG.md:18]: If it's fixing a bug it's a bug fix?

**Review by palma21 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+* Konnectivity rollout will continue in May 2022 and is expected to complete by end of May.
+* Update your AKS labels to the recommended substitutions before deprecation after the Kubernetes v1.24 release. See more information on label deprecations and how to update your labels in the [Use labels in an AKS cluster](https://docs.microsoft.com/azure/aks/use-labels) documentation.
+
+### Release notes
+* Public preview
+  * The `aks-preview` Azure CLI extension (version 0.5.66+) now supports running `az aks update -g <resourceGroup> -n <clusterName>` without any optional arguments. This will perform an update operation without performing any changes, which can recover a cluster stuck in a failed provisioning state.
+  * [AKS now supports updating kubelet on node pools to use a new or changed user-assigned managed identity.](https://docs.microsoft.com/azure/aks/use-managed-identity#update-an-existing-cluster-using-kubelet-identity-preview)
+* Behavioral changes
```
Comment (phealy) [inline CHANGELOG.md:17]:
```suggestion
* Bug fixes
  * A check for incorrect `outboundType` when deploying a cluster into a subnet with an attached NAT Gateway that was applying on update operations has been corrected to apply only to cluster creates.
```

**Review by phealy — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (phealy) [reply CHANGELOG.md:19]: Changed the header when I added the bugfix item.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+  * [AKS now supports updating kubelet on node pools to use a new or changed user-assigned managed identity.](https://docs.microsoft.com/azure/aks/use-managed-identity#update-an-existing-cluster-using-kubelet-identity-preview)
+* Bug fixes
+  * A check for incorrect `outboundType` when deploying a cluster into a subnet with an attached NAT Gateway that was applying on update operations has been corrected to apply only to cluster creates.
+    * Fixes a bug with the AKS-EnableDualStack preview feature that would delete managed outbound IPv6 IPs if updating the cluster with a version of the API before the dual-stack parameters were added.
+* Component Updates
+    *Azure File CSI driver has been updated to [v1.6](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.16.0)
+    *AKS Ubuntu 18.04 image updated to [AKSUbuntu-1804-2022.05.04](vhd-notes/aks-ubuntu/AKSUbuntu-1804/2022.05.04.txt).
```
Comment (phealy) [inline CHANGELOG.md:22]:
```suggestion
  * AKS Ubuntu 18.04 image updated to [AKSUbuntu-1804-2022.05.04](vhd-notes/aks-ubuntu/AKSUbuntu-1804/2022.05.04.txt).
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+
+### Release notes
+* Public preview
+  * The `aks-preview` Azure CLI extension (version 0.5.66+) now supports running `az aks update -g <resourceGroup> -n <clusterName>` without any optional arguments. This will perform an update operation without performing any changes, which can recover a cluster stuck in a failed provisioning state.
+  * [AKS now supports updating kubelet on node pools to use a new or changed user-assigned managed identity.](https://docs.microsoft.com/azure/aks/use-managed-identity#update-an-existing-cluster-using-kubelet-identity-preview)
+* Bug fixes
+  * A check for incorrect `outboundType` when deploying a cluster into a subnet with an attached NAT Gateway that was applying on update operations has been corrected to apply only to cluster creates.
+    * Fixes a bug with the AKS-EnableDualStack preview feature that would delete managed outbound IPv6 IPs if updating the cluster with a version of the API before the dual-stack parameters were added.
+* Component Updates
+    *Azure File CSI driver has been updated to [v1.6](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.16.0)
```
Comment (phealy) [inline CHANGELOG.md:21]:
```suggestion
  * Azure File CSI driver has been updated to [v1.6](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.16.0)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,25 @@
...
+
+### Release notes
+* Public preview
+  * The `aks-preview` Azure CLI extension (version 0.5.66+) now supports running `az aks update -g <resourceGroup> -n <clusterName>` without any optional arguments. This will perform an update operation without performing any changes, which can recover a cluster stuck in a failed provisioning state.
+  * [AKS now supports updating kubelet on node pools to use a new or changed user-assigned managed identity.](https://docs.microsoft.com/azure/aks/use-managed-identity#update-an-existing-cluster-using-kubelet-identity-preview)
+* Bug fixes
+  * A check for incorrect `outboundType` when deploying a cluster into a subnet with an attached NAT Gateway that was applying on update operations has been corrected to apply only to cluster creates.
+    * Fixes a bug with the AKS-EnableDualStack preview feature that would delete managed outbound IPv6 IPs if updating the cluster with a version of the API before the dual-stack parameters were added.
```
Comment (phealy) [inline CHANGELOG.md:19]:
```suggestion
  * Fixes a bug with the AKS-EnableDualStack preview feature that would delete managed outbound IPv6 IPs if updating the cluster with a version of the API before the dual-stack parameters were added.
```

**Review by phealy — APPROVED** [review]
Fixing indentation.

**Pull request closed without merging**